Archipelago_Gifting.cpp

```diff
@@ -227,10 +203,42 @@ void handleGiftAPISetReply(AP_SetReply reply) {
                         }
                         if (found) break;
                     }
-                    if (!found) AP_RejectGift(gift.ID);
+                    if (!found) {
+                        giftIdsToReject.insert(gift.ID);
+                    }
                 }
+
+                AP_RejectGift(giftIdsToReject);
             }
         }
+    } else if (reply.key.rfind("GiftBoxes;", 0) == 0) {
+        int team = std::stoi(reply.key.substr(10)); //10 = length of "GiftBoxes;"
+        Json::Value json_data;
+        reader.parse(*(std::string*)reply.value, json_data);
+        std::set<std::string> foundPlayersNames;
```

Comment:
In the setnotify we always get the full motherbox info, so instead of keeping track of players found, updating existing ones and removing ones that were not present, just clear map_players_to_giftbox and recreate it.

Reply:
No thats exactly how you introduce a race condition, as during the time you re-build the map the data is invallid

Reply:
The data (for specific players) is invalid in between anyway. At this point, you probably need to protect access to map_players_to_giftbox using a mutex. If thats not something you want to do, I can do it later and consider this out of scope for this PR